internal/commands/untrust_builder.go

```diff
@@ -0,0 +1,56 @@
+package commands
+
+import (
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/buildpacks/pack/internal/config"
+	"github.com/buildpacks/pack/internal/style"
+	"github.com/buildpacks/pack/logging"
+)
+
+func UntrustBuilder(logger logging.Logger, cfg config.Config) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "untrust-builder <builder-name>",
+		Short: "Stop trusting builder",
+		Long:  "Stop trusting builder.\n\nWhen building with this builder, all lifecycle phases will be no longer be run in a single container using the builder image.",
+		Args:  cobra.MaximumNArgs(1),
+		RunE: logError(logger, func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 || args[0] == "" {
+				logger.Infof("Usage:\n\t%s\n", cmd.UseLine())
+				return nil
+			}
+
+			builderName := args[0]
+			existingBuilders := cfg.TrustedBuilders
+			cfg.TrustedBuilders = []config.TrustedBuilder{}
+			for _, builder := range existingBuilders {
+				if builder.Name == builderName {
+					continue
+				}
+
+				cfg.TrustedBuilders = append(cfg.TrustedBuilders, builder)
+			}
```

Comment:
Can we change `cfg.TrustedBuilders` to be a hashset?
If we will do so, we won't need to go over all of the trusted builders in order to search for the builder that we want to untrust. Instead, we'll be able to do it instantly.
Maybe it's a bit of an overkill if we think there shouldn't be too many trusted builder but I don't think it should be too complicated to make this change.
If not a hashset, we can use a list (or other data structure) so we'll be able to stop the for loop when we find the builder that we want to untrust, instead of going over all of the existing builders.

Reply:
The concern I'd have with the hashset approach is that this data structure is a representation of the config file. We'd lose all of the toml encoding & decoding magic we get for free right now. I don't imagine this list becoming unmanageable and while iterating through the whole list is not the most efficient approach, it's also unlikely to be noticeably slow.

Reply:
I agree that it's ok for now since the trusted builder array is probably not that big, but I always prefer to be as proficient as possible so we won't have any problem with the performance in the future.
Can you please elaborate on what is the magic you're talking about and what we would need to do in order to change the trusted builder data structure?

Reply:
The "magic" is just [`toml.Decode`](https://github.com/buildpacks/pack/blob/main/internal/config/config.go#L50) and [`toml.Encode`](https://github.com/buildpacks/pack/blob/main/internal/config/config.go#L68) and the human readable format of the resultant toml file. I'm not sure how these would handle a `map[TrustedBuilder]interface{}` I'm not even sure if toml is capable of expressing that pattern.

Reply:
Interesting. I'm wondering if it's possible to do it with a list or a hashset.
Of course it doesn't have to be part of this PR.